shared/Rakefile

```diff
@@ -65,17 +65,31 @@ end
 
 require "rake/testtask"
 desc "Run functional tests for all custom protos"
-Rake::TestTask.new do |t|
-  t.libs << "test"
-  t.test_files = FileList["test/**/*_test.rb"]
-  t.warning = false
+task :test do
+  Rake::Task["test:showcase"].invoke
 end
 namespace :test do
   desc "Run functional tests for showcase protos"
   Rake::TestTask.new :showcase do |t|
-    t.libs << "test"
+    require "tmpdir"
+
+    client_lib = Dir.mktmpdir
+    protoc_cmd = [
+      "grpc_tools_ruby_protoc",
+      "--proto_path=api-common-protos",
+      "--proto_path=protos/showcase",
+      "--ruby_out=#{client_lib}",
+      "--grpc_out=#{client_lib}",
+      "--ruby_gapic_out=#{client_lib}",
+      "protos/showcase/v1alpha3/echo.proto"
+    ].join " "
+    sh protoc_cmd
+
+    t.libs << ["test",  client_lib]
     t.test_files = FileList["test/showcase/**/*_test.rb"]
     t.warning = false
+
+    # TODO: delete the temporary dir?
```

Comment:
```suggestion
    Minitest.after_run do
      FileUtils.remove_dir client_lib, true
    end
```

Reply:
I added this when I moved it to the helper. Tx!